pymc3/data.py

```diff
@@ -81,3 +88,61 @@ def __eq__(self, other):
 
     def __hash__(self):
         return hash(id(self))
+
+
+class DataSampler(object):
+    """
+    Convenient picklable data sampler for minibatch inference.
+
+    This generator can be used for passing to pm.generator
+    creating picklable theano computational grapf
+
+    Parameters
+    ----------
+    data : array like
+    batchsize : sample size over zero axis
+    seed : int for numpy random generator
+    dtype : str representing dtype
+
+    Usage
+    -----
+    >>> import pickle
+    >>> from functools import partial
+    >>> np.random.seed(42) # reproducibility
+    >>> pm.set_tt_rng(42)
+    >>> data = np.random.normal(size=(1000,)) + 10
+    >>> minibatches = DataSampler(data, batchsize=50)
+    >>> with pm.Model():
+    ...     sd = pm.Uniform('sd', 0, 10)
+    ...     mu = pm.Normal('mu', sd=10)
+    ...     obs_norm = pm.Normal('obs_norm', mu=mu, sd=sd,
+    ...                          observed=minibatches,
+    ...                          total_size=data.shape[0])
+    ...     adam = partial(pm.adam, learning_rate=.8) # easy problem
+    ...     approx = pm.fit(10000, method='advi', obj_optimizer=adam)
+    >>> new = pickle.loads(pickle.dumps(approx))
+    >>> new #doctest: +ELLIPSIS
```

Comment:
I was doing some tests with Doctest utils, this mean that some part of output `...` can change and it is not a problem